cppfd/CMakeLists.txt

```diff
@@ -40,10 +45,22 @@ add_executable(allTests allTests.cpp)
 message(STATUS "VTK libraries: ${VTK_LIBRARIES}")
 
 # Link targets to libraries
-target_link_libraries(mesh_chunk Kokkos::kokkos ${VTK_LIBRARIES})
-target_link_libraries(parallel_mesh Kokkos::kokkos ${VTK_LIBRARIES})
+if (${USE_MPI} AND MPI_FOUND)
+    MESSAGE(STATUS "MPI found. Enabling MPI.")
+    target_link_libraries(mesh_chunk MPI::MPI_CXX)
+    target_link_libraries(parallel_mesh MPI::MPI_CXX)
+    target_link_libraries(boundary_conditions MPI::MPI_CXX)
+    target_link_libraries(solver MPI::MPI_CXX)
+    target_link_libraries(test MPI::MPI_CXX)
+    target_link_libraries(parallel_test MPI::MPI_CXX)
+    target_link_libraries(allTests MPI::MPI_CXX)
+elseif(${USE_MPI} AND NOT MPI_FOUND)
+    MESSAGE(WARNING "MPI not found. Disabling MPI.")
```

Comment:
As long as MPI is marked as `REQUIRED`, CMake should fail with adequate error in the `find_package` step.